lib/src/models/comment_impl.dart

```diff
@@ -136,77 +139,81 @@ class MoreComments extends RedditBase {
   }
 }
 
-/// A class which represents a single Reddit comment.
-class Comment extends UserContent with InboxableMixin {
-  static final RegExp _commentRegExp = new RegExp(r'{id}');
-  Submission _submission;
-  CommentForest _replies;
+class Comment extends CommentRef {
+  // CommentModeration get mod; // TODO(bkonyi): implement
 
   /// Returns true if the current [Comment] is a top-level comment. A [Comment]
   /// is a top-level comment if its parent is a [Submission].
-  Future<bool> get isRoot async {
-    final parentIdType = (await _parentId).split('_')[0];
+  bool get isRoot {
+    final parentIdType = parentId.split('_')[0];
```

Comment:
Looked into the docs and it says that this is the goto strategy:  "you might call an async function from your function".

Guessing that's what you're doing here but how come you don't have to handle the parentId as a Future type?